### DISCUSSION TITLE
Java

### DISCUSSION THREAD

**Discussion by Zuba43** [community/community#141879] [Programming Help]
### Body

I have all my Java swing project done and created an executable for windows OS. How do I create one for a Linux based Os?

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by EbbeWertzStudentUH** [comment]
Java is a cross platform language, meaning that an executable jar file generally runs on any machine. Offcourse when using GUI libraries such as swing, then they might use OS specific bindings since they need low level communication to the graphics drivers.

So to make your application fully cross platform, you only need to consider specific dependencies (which in your case is probably only swing).
When a library has multiple OS versions, it is usually marked by the `classifier` tag. Depending on your build system, it may look like this:
 - Maven:
 `<dependency>
            <groupId>...</groupId>
            <artifactId>...</artifactId>
            <version>...</version>
            <classifier>win</classifier>
        </dependency>`
 - gradle:
 `implementation group: '...', name: '...', version: ..., classifier: 'win'`
 
> You can swap the `win` here for `mac` or `linux`

Now if your question also regards native executables like `.exe`, that just depends on the packager. For each operating system there exist packager programs to package an executable jar file into a native executable.
